### PR TITLE
FIT-355-Standardized the Jenkins File and added the Smoke Test

### PR DESCRIPTION
**Pull request by AVandana-OSI**
### Jira Story
https://osi-cwds.atlassian.net/browse/FIT-355

## Description
Standardized the Jenkins File and removed the deploy pre-int stage and replaced with trigger release pipeline and it triggers different Jenkins on Management server, the old intake Jenkins job will be disabled. You can find the new Jenkins job here http://jenkins.mgmt.cwds.io:8080/job/PreInt-Integration/job/deploy-intake-app/, this Job handles to deploy to pre-int, update manifest, rune smoke test also for Integration too. 

**_If any questions_**: Please contact @AVandana-OSI or FIT Team

### CODE REVIEW

**Comment by AVandana-OSI** [comment]
retest this please

**Comment by AVandana-OSI** [comment]
retest this please

File: Jenkinsfile [hunk 1]
```diff
@@ -1,165 +1,305 @@
...
+      cleanWs()
     }
+  }
+}
 
-    if (branch == 'origin/master') {
-      slackAlertColor = successColor
-      slackMessage = "${pipelineStatus}: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]' completed for branch '${branch}' (${env.BUILD_URL})"
+def slackNotification(pipelineStatus) {
+  slackAlertColor = successColor
```
Comment (edgibbs) [inline Jenkinsfile]: We should probably pull slackNotification into a shared library, but obviously not in scope for this.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,165 +1,305 @@
...
+  }
+}
 
-  } catch (e) {
-    pipelineStatus = 'FAILED'
-    currentBuild.result = 'FAILURE'
-    throw e
+def reports() {
+  stage ('Reports') {
+    step([$class: 'JUnitResultArchiver', testResults: '**/reports/*.xml'])
```
Comment (edgibbs) [inline Jenkinsfile]: could try 

```groovy
JUnitResultArchiver testResults: '**/reports/*.xml'
```

File: Jenkinsfile [hunk 3]
```diff
@@ -1,165 +1,305 @@
...
+    ])
 
-    if (branch != 'origin/master') {
-      stage('Verify SemVer Label') {
-        checkForLabel("intake")
-      }
+    try {
+      buildingTestBench()
+      lintTest()
+      veriftySemVerLabel()
```
[the remark this answers is not in the document]
Comment (AVandana-OSI) [reply Jenkinsfile]: Done

File: Jenkinsfile [hunk 4]
```diff
@@ -1,165 +1,305 @@
...
 
-      stage('Release') {
-        curStage = 'Release'
-        withEnv(["BUILD_DATE=${buildDate}","VERSION=${VERSION}","VCS_REF=${VCS_REF}"]) {
-          sh 'make release'
-        }
-      }
+def incrementTag() {
+  stage('Increment Tag') {
+    VERSION = newSemVer()
```
[the remark this answers is not in the document]
Comment (AVandana-OSI) [reply Jenkinsfile]: Done

File: Jenkinsfile [hunk 5]
```diff
@@ -1,165 +1,305 @@
...
+    VCS_REF = sh(
```
[the remark this answers is not in the document]
Comment (AVandana-OSI) [reply Jenkinsfile]: done

File: Jenkinsfile (hunk 1 above)
Comment (AVandana-OSI) [reply Jenkinsfile]: Sure, we can create a story regarding that

**Comment by AVandana-OSI** [comment]
retest this please

File: Jenkinsfile [hunk 6]
```diff
@@ -1,165 +1,305 @@
...
+def scmInfo
+def branch
+def curStage = 'Start'
+def pipelineStatus = 'SUCCESS'
+def successColor = '11AB1B'
+def failureColor = '#FF0000'
+SimpleDateFormat dateFormatGmt = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+def buildDate = dateFormatGmt.format(new Date())
+def docker_credentials_id = '6ba8d05c-ca13-4818-8329-15d41a089ec0'
+GITHUB_CREDENTIALS_ID = '433ac100-b3c2-4519-b4d6-207c029a103b'
```
[the remark this answers is not in the document]
Comment (AVandana-OSI) [reply Jenkinsfile]: If I def this the credentials are unable to pick at 
```
ws {
      git branch: "master", credentialsId: GITHUB_CREDENTIALS_ID, url: DE_ANSIBLE_GITHUB_URL
```

**Comment by AVandana-OSI** [comment]
retest this please

**Comment by AVandana-OSI** [comment]
retest this please

**Pull request merged**